lib/new_relic/agent/instrumentation/dynamodb.rb

```diff
@@ -0,0 +1,25 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative 'dynamodb/instrumentation'
+require_relative 'dynamodb/chain'
+require_relative 'dynamodb/prepend'
+
+DependencyDetection.defer do
+  named :dynamodb
+
+  depends_on do
+    defined?(Aws::DynamoDB::Client)
+  end
+
+  executes do
+    NewRelic::Agent.logger.info('Installing dynamodb instrumentation')
+
+    if use_prepend?
+      prepend_instrument Aws::DynamoDB::Client, NewRelic::Agent::Instrumentation::Dynamodb::Prepend
+    else
+      chain_instrument NewRelic::Agent::Instrumentation::Dynamodb::Chain
+    end
```

Comment:
nit: AWS uses `DynamoDB` and we use `Dynamodb`. Would you be open to updating our classes to use the same capitalization style? (This is probably related to a limitation of the generator I should look into...)

Reply:
yeah, i can't remember exactly what now, but i did end up generating a couple of times bc of weirdness with the capitalization, this one ended up looking most normal so i just did this way, but i can go in and manually change it

Reply:
Thank you!